src/github3/repos/invitation.py

```diff
@@ -0,0 +1,99 @@
+# -*- coding: utf-8 -*-
+"""Invitation related logic."""
+from __future__ import unicode_literals
+
+from json import dumps
+
+from .. import models
+from .. import users
+
+from ..decorators import requires_auth
+
+
+class Invitation(models.GitHubCore):
+    """Representation of an invitation to collaborate on a repository.
+
+    .. attribute:: created_at
+
+        TODO
+
+    .. attribute:: html_url
+
+        TODO
+
+    .. attribute:: id
+
+        TODO
+
+    .. attribute:: invitee
+
+        TODO
+
+    .. attribute:: inviter
+
+        TODO
+
+    .. attribute:: permissions
+
+        TODO
+
+    .. attribute:: repository
+
+        TODO
+
+    .. attribute:: url
+
+        TODO
+    """
+
+    class_name = 'Invitation'
+    allowed_permissions = frozenset(['admin', 'read', 'write'])
+
+    def _update_attributes(self, invitation):
+        from . import repo
+        self.created_at = self._strptime(invitation['created_at'])
+        self.html_url = invitation['html_url']
+        self.id = invitation['id']
+        self.invitee = users.ShortUser(invitation['invitee'], self)
+        self.inviter = users.ShortUser(invitation['inviter'], self)
+        self.permissions = invitation['permissions']
+        self.repository = repo.ShortRepository(invitation['repository'], self)
+        self.url = invitation['url']
+
+    def _repr(self):
+        return '<Invitation [{0}]>'.format(self.repository.full_name)
+
+    @requires_auth
+    def delete(self):
+        """Delete this invitation.
+
+        :returns:
+            True if successful, False otherwise
+        :rtype:
+            bool
+        """
+        url = self._build_url(
+            'invitations', self.id, base_url=self.repository.url)
+        return self._boolean(self._delete(url), 204, 404)
+
+    @requires_auth
+    def update(self, permissions):
+        """Update this invitation.
+
+        :param str permissions:
+            (required), the permissions that will be granted by this invitation
+            once it has been updated. Options: 'admin', 'read', 'write'
+        :returns:
+            The updated invitation
+        :rtype:
+            :class:`~github3.repos.invitation.Invitation`
+        """
+        if permissions not in self.allowed_permissions:
+            raise ValueError("'permissions' must be one of {0}".format(
+                ', '.join(sorted(self.allowed_permissions))
+            ))
+        url = self._build_url(
+            'invitations', self.id, base_url=self.repository.url)
```

Comment:
This URL-building logic is repeated in `delete` and `update`. Should I do it once in `_update_attributes` and assign to `self._api` ? (The `url` returned by the API is sadly not the right one.)

Reply:
The crux of the issue is that there's two API URL, one used by the inviter to cancel/update an invitation, the other used by the inviter to accept/decline. WDYT of computing two private properties called `_invitee_api` and `_inviter_api` in `_update_attributes` ?

Reply:
We build/rebuild URLs in a few places. I don't think it's the worst thing to have these work this way.